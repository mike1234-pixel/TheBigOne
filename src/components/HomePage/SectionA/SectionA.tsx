import React from "react";
import "./SectionA.scss";
import PropTypes from "prop-types";

const SectionA = (props) => {
  const loremText =
    "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.";

  return (
    <div className="section-a grid-container">
      <div className="grid-item grid-item-one">
        <h2>item</h2>
        <p>{loremText}</p>
      </div>

      <div className="grid-item grid-item-two">
        <h2>item</h2>
        <p>{loremText}</p>
      </div>

      <div className="grid-item grid-item-three">
        <h2>item</h2>
        <p>{loremText}</p>
      </div>
    </div>
  );
};

export default SectionA;

SectionA.propTypes = {
  darkMode: PropTypes.bool,
};
