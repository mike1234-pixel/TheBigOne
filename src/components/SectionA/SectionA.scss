// for smooth transition between two colors can use css transition property.
// for smooth transition between color and image or image and image, use react transition group.

.section-a {
  width: 100vw;
  height: 75vh;
  background-color: #fbfbfb;
  color: #000;
}

.grid-container {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: 100%;
  grid-gap: 30px;
  padding: 5%;
}

.grid-item {
  height: 100px;
  width: 100px;

  &-one {
    grid-column: 1/2;
    background-color: red;
    transition: 1s;
    &:hover,
    &:active {
      background-color: purple;
      cursor: pointer;
    }
  }
  &-two {
    grid-column: 2/3;
    background-color: pink;
    transition: 1s;
    &:hover,
    &:active {
      background-image: url(../Carousel/carouselImages/img2.webp);
      cursor: pointer;
    }
  }
  &-three {
    grid-column: 3/4;
    background-image: url(../Carousel/carouselImages/img4.webp);
    &:hover,
    &:active {
      background-image: url(../Carousel/carouselImages/img3.webp);
      cursor: pointer;
    }
  }
}
