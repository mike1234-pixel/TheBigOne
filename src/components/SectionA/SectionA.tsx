import React from "react";
import "./SectionA.scss";

const SectionA = () => {
  return (
    <div className="section-a grid-container">
      <div className="grid-tem grid-item-one">item</div>
      <div className="grid-tem grid-item-two">item</div>
      <div className="grid-tem grid-item-three">item</div>
    </div>
  );
};

export default SectionA;

// hover over text to reveal a picture
