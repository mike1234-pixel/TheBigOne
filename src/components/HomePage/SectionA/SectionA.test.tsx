import React from "react";
import SectionA from "./SectionA";
import renderer from "react-test-renderer";

// snapshot test
const darkMode = true;

it("renders correctly", () => {
  const tree = renderer.create(<SectionA darkMode={darkMode} />).toJSON();
  expect(tree).toMatchSnapshot();
});
