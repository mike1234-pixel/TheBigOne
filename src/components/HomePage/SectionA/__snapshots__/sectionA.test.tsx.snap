// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="section-a grid-container"
>
  <div
    className="grid-item grid-item-one"
  >
    <h2>
      item
    </h2>
    <p>
      Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
    </p>
  </div>
  <div
    className="grid-item grid-item-two"
  >
    <h2>
      item
    </h2>
    <p>
      Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
    </p>
  </div>
  <div
    className="grid-item grid-item-three"
  >
    <h2>
      item
    </h2>
    <p>
      Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
    </p>
  </div>
</div>
`;
